### PR TITLE
Fix #6 - Allow early ending table/createReadStream

### LINKED ISSUE #6: Bigtable this.end() does not stop the stream?
_From @arbesfeld on May 2, 2017 14:15_

Using Bigtable 0.9.1 - I seem to be getting more than 1 result after calling `this.end()`. Is this expected behavior?

_Copied from original issue: GoogleCloudPlatform/google-cloud-node#2271_

### PR DESCRIPTION
**Pull request by jiren**
Fixed data transformation stream to stop processing data on `end` event
Still, need to fix GRPC stream closing.

Fixes #6 

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by jiren** [comment]
To fix GRPC stream ending need to call `abort` function on `retryRequest` stream.

File Ref: [common-grpc/src/service.js:L398](https://github.com/GoogleCloudPlatform/google-cloud-node/blob/master/packages/common-grpc/src/service.js#L398)

**Comment by sduskis** [comment]
This should fix issue #6

**Comment by sduskis** [comment]
This looks good to me.  @stephenplusplus, what do you think?

**Comment by stephenplusplus** [comment]
I think work is still pending on the PR, judging from the comment and this line in the initial post:

> Still, need to fix GRPC stream closing.

**Comment by jiren** [comment]
GRPC stream ending need to fix in [common-grpc/src/service.js:L398](https://github.com/GoogleCloudPlatform/google-cloud-node/blob/master/packages/common-grpc/src/service.js#L398). I will create pull request in `common-grpc` repo.

**Comment by stephenplusplus** [comment]
Sorry, I misunderstood. The fix looks good, but I left one edit request for the test.

File: system-test/bigtable.js [hunk 1]
```diff
@@ -668,6 +668,21 @@ describe('Bigtable', function() {
           });
       });
 
+      it('should end stream early', function(done) {
+        var rows = [];
+
+        TABLE.createReadStream()
```
[the remark this answers is not in the document]
Comment (jiren) [reply system-test/bigtable.js]: Sure, I will update test

[the remark this answers is not in the document]
Comment (jiren) [reply system-test/bigtable.js]: @stephenplusplus Updated system test

**Comment by stephenplusplus** [comment]
Thanks!

**Pull request merged**